pkg/helpers/crd.go

```diff
@@ -23,11 +23,13 @@ func CreateGitRepo(obj metav1.Object, gvk schema.GroupVersionKind, template *syn
 
 	tenantName := obj.GetName()
 	tenantNamespace := obj.GetNamespace()
-	if tenantRef != nil {
+	if tenantRef != nil || tenantName == "" {
 		tenantName = tenantRef.Name
```

Comment:
`tenantRef` could be `nil` here. Should probably be AND instead of OR?

Reply:
Simplified the logic.